test/e2e/util/mongotester/mongotester.go

```diff
@@ -254,21 +227,19 @@ func (m *Tester) connectivityCheck(shouldSucceed bool, opts ...OptionApplier) fu
 	}
 }
 
-func (m *Tester) WaitForRotatedCertificate() func(*testing.T) {
+func (m *Tester) WaitForRotatedCertificate(mdb mdbv1.MongoDBCommunity, initialCertSerialNumber *big.Int) func(*testing.T) {
 	return func(t *testing.T) {
-		// The rotated certificate has serial number 2
-		expectedSerial := big.NewInt(2)
-
-		tls, err := getClientTLSConfig()
+		tls, err := getClientTLSConfig(mdb)
 		assert.NoError(t, err)
 
 		// Reject all server certificates that don't have the expected serial number
 		tls.VerifyPeerCertificate = func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
 			cert := verifiedChains[0][0]
-			if expectedSerial.Cmp(cert.SerialNumber) != 0 {
-				return errors.Errorf("expected certificate serial number %s, got %s", expectedSerial, cert.SerialNumber)
+			if initialCertSerialNumber.Cmp(cert.SerialNumber) == 0 {
+				return errors.Errorf("certificate serial number has not changed: %s", cert.SerialNumber)
+			} else {
+				return nil
 			}
```

Comment:
redundant else clause, we can just `return nil` here

Reply:
removed